listener/kcp/manager_factory.go

```diff
@@ -8,19 +8,29 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	kcpctrl "sigs.k8s.io/controller-runtime/pkg/kcp"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/openmfp/kubernetes-graphql-gateway/common/config"
 )
 
 type ManagerFactory struct {
-	IsKCPEnabled bool
+	AppConfig *config.Config
```

Comment:
We need 3 config fields, let me know if you are okay with passing full config, I choose that to simplify amount of params in virtualWorkspaceConfigFromCfg func

Reply:
You are right, this field must be private since it is supposed to be used by manager only.
Also, removed pointer ref.